### PR TITLE
Introduce QDQ transformer fusion tools for ordered quantized ops

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: This change introduces 4 fusions used to fuse QDQ subgraphs found in QAT BERT (Tested on a HuggingFace PyTorch QAT BERT model):
                         (1) QOrdereGelu
                         (2) QOrderedLayerNormalization 
                         (3) QOrderedMatMul
                         (4) QOrderedAttention

### CODE REVIEW

File: onnxruntime/python/tools/transformers/fusion_utils.py [hunk 1]
```diff
@@ -83,6 +83,35 @@ def check_node_attribute(node, attribute_name: str, expected_value, default_valu
...
+           It is a good candidate for fusion if:
+           (1) The Q/DQ node is for per-tensor quantization (per-axis quantization is not supported)
+           (2) The Q/DQ node should have constant scale
+           (3) The Q/DQ node should have a zero point of 0
+        Args:
+            node (NodeProto): a Q/DQ node to check
+        Returns:
+            bool: whether the check is passed or not
+        """
+        if not(node.op_type == "QuantizeLinear" or node.op_type == "DequantizeLinear"):
```
Comment (justinchuby) [inline onnxruntime/python/tools/transformers/fusion_utils.py:98]:
```suggestion
        if not node.op_type in {"QuantizeLinear", "DequantizeLinear"}:
```

File: onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+from onnx_model import OnnxModel
+
+logger = getLogger(__name__)
+
+
+class FusionQOrderedLayerNormalization(Fusion):
+    def __init__(self, model: OnnxModel):
+        super().__init__(model, "QOrderedLayerNormalization", "LayerNormalization")
+
+    def fuse(self, node, input_name_to_nodes: Dict, output_name_to_node: Dict):
```
Comment (justinchuby) [inline onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py:20]: Consider using Mapping for the annotation, here and elsewhere

```suggestion
    def fuse(self, node, input_name_to_nodes: Mapping[str, Any], output_name_to_node: Mapping[str, Any]):
```

If we know how the nodes are typed, I would also replace Any with that

-----

Use more generic types for input parameters and specific types for return values. For example, a function may take a `Sequence` and return a `List`.

Diagram for reference: https://gist.github.com/justinchuby/4021cebe9e093f636759a88de325c85f

File: onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation.  All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from logging import getLogger
```
Comment (justinchuby) [inline onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py:5]: I would add a brief docstring to summarize the module:

https://google.github.io/styleguide/pyguide.html#382-modules

File: onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py (hunk 2 above)
Comment (hariharans29) [reply onnxruntime/python/tools/transformers/fusion_qordered_layernorm.py:20]: Thanks for this education. I will read more about this.

**Review by zhanghuanrong — APPROVED** [review]

File: onnxruntime/python/tools/transformers/onnx_model_bert.py [hunk 4]
```diff
@@ -14,6 +14,10 @@
 from fusion_gelu_approximation import FusionGeluApproximation
 from fusion_layernorm import FusionLayerNormalization, FusionLayerNormalizationTF
 from fusion_options import FusionOptions
+from fusion_qordered_attention import FusionQOrderedAttention
```
Comment (wangyems) [inline onnxruntime/python/tools/transformers/onnx_model_bert.py:17]: lgtm

**Review by zhanghuanrong — DISMISSED** [review]
LGTM

**Review by yufenglee — APPROVED** [review]

**Pull request merged**